firedrake/slate/slac/optimise.py

```diff
@@ -69,7 +69,8 @@ def _push_block(expr, self, indices):
 @_push_block.register(Transpose)
 def _push_block_transpose(expr, self, indices):
     """Indices of the Blocks are transposed if Block is pushed into a Transpose."""
-    return Transpose(*map(self, expr.children, repeat(indices[::-1]))) if indices else expr
+    return (Transpose(*map(self, expr.children, repeat(indices[::-1])))
+            if indices else Transpose(*map(self, expr.children, repeat(indices))))
```

Comment:
In general I need this `repeat` due to the map. Imagine a node has two children then I want to do `self(child, indices)` for both children. `Map` needs to be passed the extra arguments once for each child. I could do` (indices, indices)` instead of `repeat(indices)` in that case, but repeat has the advantage that it only creates the memory for `indices` once. Now `Transpose` has only one child, so I could do `(indices,)` instead of `repeat(indices)` but I kept it so it's aligned to other cases. So long answer in short `repeat(indices)` would be a generator that only contains `()`

Reply:
OK thanks for explaining.